### PR TITLE
Feature/msdesc parser

### PR DESCRIPTION
**Pull request by saramaenza**

### CODE REVIEW

**Comment by szenzaro** [comment]
Solve the linting error that make the build fail
```
ERROR: /home/travis/build/evt-project/evt-viewer-angular/src/app/models/evt-models.ts:390:2 - file should end with a newline
ERROR: /home/travis/build/evt-project/evt-viewer-angular/src/app/services/xml-parsers/index.ts:18:139 - trailing whitespace
ERROR: /home/travis/build/evt-project/evt-viewer-angular/src/app/services/xml-parsers/index.ts:20:103 - trailing whitespace
ERROR: /home/travis/build/evt-project/evt-viewer-angular/src/app/services/xml-parsers/msdesc-parser.ts:226:2 - file should end with a newline```

File: src/app/models/evt-models.ts [hunk 1]
```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
...
+export class MsIdentifier extends GenericElement {
+    id?: string;
+    settlement?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when settlement is handled
+    repository?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when repository is handled
+    idno?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when idno is handled
+    institution?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when institution is handled
+    altIdentifier?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when altIdentifier is handled
+    msName?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when msName is handled
+    country?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when country is handled
+    region?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when region is handled
```
Comment (szenzaro) [inline src/app/models/evt-models.ts:353]: Try to give a definition for each field

File: src/app/models/evt-models.ts [hunk 2]
```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
...
+}
+
+export class MsContents extends GenericElement {
+    summary?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when summary is handled
+    msItem?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when msItem is handled
+    msItemStruct?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when msItemStruct is handled
```
Comment (szenzaro) [inline src/app/models/evt-models.ts:359]: Try to give a definition for each field

File: src/app/models/evt-models.ts [hunk 3]
```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
...
+    objectDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when objectDesc is handled
+    bindingDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when bindingDesc is handled
+    decoDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when decoDesc is handled
+    handDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when handDesc is handled
+    accMat?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when accMat is handled
+    additions?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when additions is handled
+    musicNotation?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when musicNotation is handled
+    scriptDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when scriptDesc is handled
+    sealDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when sealDesc is handled
+    typeDesc?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when typeDesc is handled
```
Comment (szenzaro) [inline src/app/models/evt-models.ts:372]: Try to give a definition for each field

File: src/app/models/evt-models.ts [hunk 4]
```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
...
+    additional?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when additional is handled
+    history?: History;
+    head?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when head is handled
+}
+
+export class History extends GenericElement {
+    acquisition?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when acquisition is handled
+    origin?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when origin is handled
+    provenance?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when provenance is handled
+    summary?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when summary is handled
```
Comment (szenzaro) [inline src/app/models/evt-models.ts:389]: Try to give a definition for each field

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        const acquisition = Array.from(xml.querySelectorAll<XMLElement>(':scope > acquisition'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when origin is handled
+        const origin = Array.from(xml.querySelectorAll<XMLElement>(':scope > origin'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when provenance is handled
+        const provenance = Array.from(xml.querySelectorAll<XMLElement>(':scope > provenance'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when summary is handled
+        const summary = Array.from(xml.querySelectorAll<XMLElement>(':scope > summary'))
```
Comment (szenzaro) [inline src/app/services/xml-parsers/msdesc-parser.ts:19]: Try to transform the xml data into typescript objects

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when scriptDesc is handled
+        const scriptDesc = Array.from(xml.querySelectorAll<XMLElement>(':scope > scriptDesc'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when sealDesc is handled
+        const sealDesc = Array.from(xml.querySelectorAll<XMLElement>(':scope > sealDesc'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when typeDesc is handled
+        const typeDesc = Array.from(xml.querySelectorAll<XMLElement>(':scope > typeDesc'))
+        .map(e => parseChildren(e, this.genericParse));
```
Comment (szenzaro) [inline src/app/services/xml-parsers/msdesc-parser.ts:68]: Try to transform the xml data into typescript objects

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+    parse(xml: XMLElement): MsContents {
+        // TODO: Add specific parser when summary is handled
+        const summary = Array.from(xml.querySelectorAll<XMLElement>(':scope > summary'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when msItem is handled
+        const msItem = Array.from(xml.querySelectorAll<XMLElement>(':scope > msItem'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when idno is handled
+        const msItemStruct = Array.from(xml.querySelectorAll<XMLElement>(':scope > msItemStruct'))
```
Comment (szenzaro) [inline src/app/services/xml-parsers/msdesc-parser.ts:100]: Try to transform the xml data into typescript objects

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when msName is handled
+        const msName = Array.from(xml.querySelectorAll<XMLElement>(':scope > msName'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when msName is handled
+        const country = Array.from(xml.querySelectorAll<XMLElement>(':scope > country'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when altIdentifier is handled
+        const region = Array.from(xml.querySelectorAll<XMLElement>(':scope > region'))
+        .map(e => parseChildren(e, this.genericParse));
```
Comment (szenzaro) [inline src/app/services/xml-parsers/msdesc-parser.ts:142]: Try to transform the xml data into typescript objects

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+        const msContentsEl = xml.querySelector<XMLElement>('scope > msContents');
+        const physDescEl = xml.querySelector<XMLElement>('scope > physDesc');
+        const msPartEl = xml.querySelector<XMLElement>('scope > msPart');
+        const historyEl = xml.querySelector<XMLElement>('scope > history');
+        // TODO: Add specific parser when additional is handled
+        const additional = Array.from(xml.querySelectorAll<XMLElement>(':scope > additional'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when head is handled
+        const head = Array.from(xml.querySelectorAll<XMLElement>(':scope > head'))
+        .map(e => parseChildren(e, this.genericParse));
```
Comment (szenzaro) [inline src/app/services/xml-parsers/msdesc-parser.ts:180]: Try to transform the xml data into typescript objects

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+    private physDescParser = createParser(PhysDescParser, this.genericParse);
+    private msPartParser = createParser(MsPartParser, this.genericParse);
+    private historyParser = createParser(HistoryParser, this.genericParse);
+
+    parse(xml: XMLElement): MsDesc {
+        const msIdentifierEl = xml.querySelector<XMLElement>('scope > msIdentifier');
+        const msContentsEl = xml.querySelector<XMLElement>('scope > msContents');
+        const physDescEl = xml.querySelector<XMLElement>('scope > physDesc');
+        const msPartEl = xml.querySelector<XMLElement>('scope > msPart');
+        const historyEl = xml.querySelector<XMLElement>('scope > history');
```
Comment (szenzaro) [inline src/app/services/xml-parsers/msdesc-parser.ts:211]: Try to transform the xml data into typescript objects

**Review by szenzaro — CHANGES_REQUESTED** [review]

File: src/app/models/evt-models.ts [hunk 11]
```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
 export class Deletion extends GenericElement {
     rend: string;
 }
+
+export class MsDesc extends GenericElement {
+    id? : string;
```
Comment (giuliac89) [inline src/app/models/evt-models.ts:336]: Why is the id optional?

File: src/app/models/evt-models.ts [hunk 12]
```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
...
+    msIdentifier: MsIdentifier;
+    msContents?: MsContents;
+    physDesc?: PhysDesc;
+    history?: History;
+    msPart?: MsPart;
+}
+
+export class MsIdentifier extends GenericElement {
+    id?: string;
```
Comment (giuliac89) [inline src/app/models/evt-models.ts:345]: Why is the id optional?

**Review by giuliac89 — CHANGES_REQUESTED** [review]

File: src/app/models/evt-models.ts (hunk 1 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:353]: I personally think it is impossible to handle in detail the definition for every field of every TEI element. At least not in a single PR. We have to stop somewhere. We should probably decide together where it is ok to stop and leave the generic definition.

File: src/app/models/evt-models.ts (hunk 2 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:359]: See above

File: src/app/models/evt-models.ts (hunk 3 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:372]: See above

File: src/app/models/evt-models.ts (hunk 1 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:353]: I personally think it is impossible to handle in detail the definition for every field of every TEI element. At least not in a single PR. We have to stop somewhere. We should probably decide together where it is ok to stop and leave the generic definition.

File: src/app/models/evt-models.ts (hunk 2 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:359]: See above

File: src/app/models/evt-models.ts (hunk 4 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:389]: See above

File: src/app/models/evt-models.ts (hunk 11 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:336]: I would also set it as mandatory and create one (maybe from the XML path) if none exists in data source

File: src/app/models/evt-models.ts (hunk 12 above)
Comment (ChiaraDipi) [reply src/app/models/evt-models.ts:345]: I would also set it as mandatory and create one (maybe from the XML path) if none exists in data source

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+        };
+    }
+}
+
+export class MsPartParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    private msIdentifierParser = createParser(MsIdentifierParser, this.genericParse);
+    private msContentsParser = createParser(MsContentsParser, this.genericParse);
+    private physDescParser = createParser(PhysDescParser, this.genericParse);
+    private msPartParser = createParser(MsPartParser, this.genericParse);
```
Comment (ChiaraDipi) [inline src/app/services/xml-parsers/msdesc-parser.ts:166]: You don't need this definition. You already are inside the `MsPartParser`. If you need to use it you can just do `this.parse()`;

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+
+        return {
+            type: MsPart,
+            class: getClass(xml),
+            content: parseChildren(xml, this.genericParse),
+            attributes: this.attributeParser.parse(xml),
+            msIdentifier: msIdentifierEl ? this.msIdentifierParser.parse(msIdentifierEl) : undefined,
+            msContents: msContentsEl ? this.msContentsParser.parse(msContentsEl) : undefined,
+            physDesc: physDescEl ? this.physDescParser.parse(physDescEl) : undefined,
+            msPart : msPartEl ? this.msPartParser.parse(msPartEl) : undefined,
```
Comment (ChiaraDipi) [inline src/app/services/xml-parsers/msdesc-parser.ts:190]: See previous comment: you can simply do `this.parse(msPartEl)`

**Review by ChiaraDipi — CHANGES_REQUESTED** [review]

File: src/app/models/evt-models.ts (hunk 11 above)
Comment (giuliac89) [reply src/app/models/evt-models.ts:336]: There is already the function getID(). I have seen that it is already used

File: src/app/models/evt-models.ts (hunk 12 above)
Comment (giuliac89) [reply src/app/models/evt-models.ts:345]: The getID() function already exists

File: src/app/models/evt-models.ts (hunk 1 above)
Comment (szenzaro) [reply src/app/models/evt-models.ts:353]: for example, `msName` can be described as a `name { type: string, value: string }` instead of an array of `GenericElement`s.
The same for `country { abbr: string, value:string }`. 
This kind of approach is suitable for a lot of these fields, and IMO are bettere definitions than `GenericElement`.

I understand that some of the fields can't be treated like this. But for the rest, I propose to try giving a definition, also because the structure of msdesc seems to hold enough structure.

File: src/app/models/evt-models.ts (hunk 2 above)
Comment (szenzaro) [reply src/app/models/evt-models.ts:359]: see above

File: src/app/models/evt-models.ts (hunk 3 above)
Comment (szenzaro) [reply src/app/models/evt-models.ts:372]: see above

File: src/app/models/evt-models.ts (hunk 4 above)
Comment (szenzaro) [reply src/app/models/evt-models.ts:389]: see above

**Comment by szenzaro** [comment]
Still some linting errors to be fixed:
```
/home/travis/build/evt-project/evt-viewer-angular/src/app/models/evt-models.ts:336:7
ERROR: 336:7   typedef-whitespace      expected nospace before colon in property-declaration
ERROR: 350:35  no-trailing-whitespace  trailing whitespace
ERROR: 392:52  no-trailing-whitespace  trailing whitespace
ERROR: 413:45  no-trailing-whitespace  trailing whitespace
ERROR: 414:16  no-trailing-whitespace  trailing whitespace
/home/travis/build/evt-project/evt-viewer-angular/src/app/services/xml-parsers/index.ts:11:1
ERROR: 11:1    max-line-length         Exceeds maximum line length of 140
/home/travis/build/evt-project/evt-viewer-angular/src/app/services/xml-parsers/msdesc-parser.ts:222:22
ERROR: 222:22  whitespace              missing whitespace
ERROR: 322:38  no-trailing-whitespace  trailing whitespace
ERROR: 328:22  no-trailing-whitespace  trailing whitespace
ERROR: 332:21  no-trailing-whitespace  trailing whitespace
ERROR: 366:1   no-trailing-whitespace  trailing whitespace
ERROR: 382:2   eofline                 file should end with a newline
```

**Comment by szenzaro** [comment]
Still some lint errors
```
Linting "evt-viewer-angular"...
/home/travis/build/evt-project/evt-viewer-angular/src/app/models/evt-models.ts:336:7
ERROR: 336:7   typedef-whitespace  expected nospace before colon in property-declaration
ERROR: 437:16  typedef-whitespace  expected onespace after colon in property-declaration
/home/travis/build/evt-project/evt-viewer-angular/src/app/services/xml-parsers/msdesc-parser.ts:449:22
ERROR: 449:22  whitespace          missing whitespace
```

File: src/app/models/evt-models.ts [hunk 15]
```diff
@@ -331,3 +331,163 @@ export class Word extends GenericElement {
...
+    n?: string;
+    class?: string;
+    defective?: boolean;
+    locus?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when locus is handled
+    author?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when author is handled
+    title?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when title is handled
+    rubric?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when rubric is handled
+    finalRubric?: Array<ParseResult<GenericElement>>; // TODO: Add specific type finalRubric defective is handled
+    bibl?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when bibl is handled
+}
```
Comment (ChiaraDipi) [inline src/app/models/evt-models.ts:411]: According to the TEI guidelines there are more elements that can be used to descrive the msItemStruct (see https://www.tei-c.org/release/doc/tei-p5-doc/en/html/MS.html#mscoit - 10.6.1 The and Elements)

File: src/app/models/evt-models.ts [hunk 16]
```diff
@@ -331,3 +331,163 @@ export class Word extends GenericElement {
...
+    docDate?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when docDate is handled
+    noteEl?: Note[];
+    locus?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when locus is handled
+    locusGrp?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when locusGrp is handled
+    bibl?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when bibl is handled
+    incipit?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when incipit is handled
+    explicit?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when explicit is handled
+    filiation?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when filiation is handled
+    gapEl?: Gap[];
+}
```
Comment (ChiaraDipi) [inline src/app/models/evt-models.ts:433]: According to the TEI guidelines there are more elements that can be used to descrive the `msItem` (see https://www.tei-c.org/release/doc/tei-p5-doc/en/html/MS.html#mscoit - 10.6.1 The <msItem> and <msItemStruct> Elements); at least add those indicated in the Manuscript Description section. Moreover you should consider also the fact that it may contain only paragraphs and no other structured element

> While neither msItem nor msItemStruct may contain untagged running text, both permit an unstructured description to be provided in the form of one or more paragraphs of text. They differ in this respect also: if paragraphs are supplied as the content of an msItem, then none of the other component elements listed above is permitted; in the msItemStruct case, however, paragraphs may appear anywhere as an alternative to any of the component elements listed above.

**Review by ChiaraDipi — CHANGES_REQUESTED** [review]
You should analyze better the indication given in the TEI guidelines for the encoding of manuscript description (https://www.tei-c.org/release/doc/tei-p5-doc/en/html/MS.html). I've added some comments to some of the models, but you should redesign them all a little better.

File: src/app/models/evt-models.ts [hunk 17]
```diff
@@ -330,3 +330,479 @@ export class Word extends GenericElement {
...
+    collection?: CollectionEl;
+    repository?: Repository;
+    region?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when region is handled
+    settlement?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when settlement is handled
+}
+
+export class Institution extends GenericElement {
+    country?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when country is handled
+    region?: Array<ParseResult<GenericElement>>; // TODO: Add specific type when region is handled
+ }
```
Comment (ChiaraDipi) [inline src/app/models/evt-models.ts:428]: Remove space before `}`

File: src/app/models/evt-models.ts [hunk 18]
```diff
@@ -330,3 +330,479 @@ export class Word extends GenericElement {
...
+
+export class Repository extends GenericElement {
+   lang?: string;
```
Comment (ChiaraDipi) [inline src/app/models/evt-models.ts:431]: Align `lang` properly

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 19]
```diff
@@ -0,0 +1,1594 @@
...
+
+export class AcquisitionParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+
+    parse(xml: XMLElement): Acquisition {
+        const attributes = this.attributeParser.parse(xml);
+        const { notBefore, notAfter } = attributes;
+        // TODO: Add specific parser when name is handled
+        const name = Array.from(xml.querySelectorAll<XMLElement>(':scope > name'))
+        .map(e => parseChildren(e, this.genericParse));
```
Comment (ChiaraDipi) [inline src/app/services/xml-parsers/msdesc-parser.ts:162]: Please auto format the entire file: all new line starting with `.map` are not properly aligned

**Review by ChiaraDipi — CHANGES_REQUESTED** [review]
I think that the parsers for this section are enough for now.
For me the PR can be approved, there are just some minor formatting changes to be done before.

**Review by giuliac89 — CHANGES_REQUESTED** [review]

**Review by giuliac89 — COMMENTED** [review]
The PR is ok for me. Only two small changes and rember to add this feature in changelog!

**Review by szenzaro — APPROVED** [review]

File: CHANGELOG.md [hunk 20]
```diff
@@ -40,6 +40,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Named entities visualisation
 - Named entities extraction
 - OpenSeadragon component with support for manifest file
+- Manuscript description data extraction
```
Comment (ChiaraDipi) [inline CHANGELOG.md:43]: Move this at the top of the list

**Review by szenzaro — APPROVED** [review]

File: src/app/services/xml-parsers/msdesc-parser.ts [hunk 21]
```diff
@@ -0,0 +1,954 @@
...
+    Origin, OrigPlace, Paragraph, PhysDesc, Provenance, RecordHist, Repository, Rubric, ScriptDesc, Seal, SealDesc, Source, Summary,
+    Support, SupportDesc, Surrogates, TypeDesc, TypeNote, Width, XMLElement,
+} from '../../models/evt-models';
+import { GapParser, GenericElemParser, LBParser, NoteParser, ParagraphParser, queryAndParseElement, queryAndParseElements } from './basic-parsers';
+import { GParser } from './character-declarations-parser';
+import { createParser, getClass, getDefaultN, getID, parseChildren, Parser } from './parser-models';
+
+function unhandledElement(xml: XMLElement, name: string) {
+    return flat(Array.from(xml.querySelectorAll<XMLElement>(`:scope > ${name}`)).map(e => parseChildren(e, this.genericParse)));
+}
```
Comment (ChiaraDipi) [inline src/app/services/xml-parsers/msdesc-parser.ts:17]: @giuliac89 I would put this in `parser-model.ts` file since it could be useful also for other parsers. What do you think?

File: src/app/utils/js-utils.ts [hunk 22]
```diff
@@ -88,3 +88,7 @@ export function getEventKeyCode(event) {
 export function normalizeUrl(url) {
     return url && url.indexOf('http') < 0 ? 'http://' + url : url;
 }
+
+export function getBoolean(bool: string){
```
Comment (szenzaro) [inline src/app/utils/js-utils.ts:92]: Rename `isBoolString` and change param name to something like `s`

File: src/app/utils/js-utils.ts [hunk 23]
```diff
@@ -47,10 +49,9 @@ export function uniqueArrayCharKeys(a: string[]) {
  * @param objects - Objects to merge
  * @returns New object with merged key/values
  */
-export function mergeDeep(...objects) {
-    const isObject = obj => obj && typeof obj === 'object';
-
-    return objects.reduce((prev, obj) => {
+export function mergeDeep<T>(...objects: T[]) { // TODO: never used, remove?
```
Comment (szenzaro) [inline src/app/utils/js-utils.ts:52]: Why these changes?

**Review by szenzaro — CHANGES_REQUESTED** [review]

File: src/app/services/xml-parsers/msdesc-parser.ts (hunk 21 above)
Comment (ChiaraDipi) [inline src/app/services/xml-parsers/msdesc-parser.ts:17]: I would put this in `parser-model.ts` file since it could be useful also for other parsers.

**Review by ChiaraDipi — CHANGES_REQUESTED** [review]

**Review by szenzaro — APPROVED** [review]

**Review by szenzaro — APPROVED** [review]

**Review by ChiaraDipi — APPROVED** [review]

**Review by giuliac89 — APPROVED** [review]
